### PR TITLE
[Account] Adding warning message to discourage username/password

### PR DESCRIPTION
**Pull request by lijinpei2008**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs [hunk 1]
```diff
@@ -337,6 +337,11 @@ public override void ExecuteCmdlet()
                 AzConfigReader.Instance?.UpdateConfig(ConfigKeys.EnableLoginByWam, false, ConfigScope.CurrentUser);
             }
 
+            if (ParameterSetName.Equals(UserWithCredentialParameterSet))
+            {
+                WriteWarning("Authentication with a username and password at the command line is strongly discouraged. Use one of the recommended authentication methods based on your requirements. For additional information, visit https://go.microsoft.com/fwlink/?linkid=2276971.");
```
Comment (msJinLei) [inline src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs:342]: Please use a resource string instead.

**Review by msJinLei — CHANGES_REQUESTED** [review]
Please describe the change in the change log

**Comment by lijinpei2008** [comment]
Add Breaking Change Notification to Change Access Token from String to Secure String
https://github.com/Azure/azure-powershell/pull/25419

File: src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs [hunk 2]
```diff
@@ -337,6 +337,12 @@ public override void ExecuteCmdlet()
                 AzConfigReader.Instance?.UpdateConfig(ConfigKeys.EnableLoginByWam, false, ConfigScope.CurrentUser);
             }
 
+            if (ParameterSetName.Equals(UserWithCredentialParameterSet))
+            {
+                string warningMessage = "Authentication with a username and password at the command line is strongly discouraged. Use one of the recommended authentication methods based on your requirements. For additional information, visit https://go.microsoft.com/fwlink/?linkid=2276971.";
+                WriteWarning(warningMessage);
```
Comment (msJinLei) [inline src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs:343]: Please add the string to Resources.resx

File: src/Accounts/Accounts/ChangeLog.md [hunk 3]
```diff
@@ -24,6 +24,8 @@
 * Updated `Microsoft.Identity.Client.NativeInterop` to fix the WAM pop window issue in elevated mode [#24967]
 * Updated the reference of Azure PowerShell Common to 1.3.98-preview.
 * Limited promotional message to interactive scenarios only
+* In `Connect-AzAccount` add warning message to discourage username/password flow
```
Comment (msJinLei) [inline src/Accounts/Accounts/ChangeLog.md:27]:
```suggestion
* In `Connect-AzAccount` add warning message to deprecate username/password flow
```

File: src/Accounts/Accounts/ChangeLog.md [hunk 4]
```diff
@@ -24,6 +24,8 @@
...
+* In `Get-AzAccessToken` add Breaking Change Notification to Change Access Token from String to Secure String
```
Comment (msJinLei) [inline src/Accounts/Accounts/ChangeLog.md:28]:
```suggestion
* In `Get-AzAccessToken` add Breaking Change notification to change AccessToken property from String to Secure String
```

File: src/Accounts/Accounts/ChangeLog.md [hunk 5]
```diff
@@ -24,6 +24,8 @@
...
```
Comment (msJinLei) [inline src/Accounts/Accounts/ChangeLog.md:29]: Move the change log just under upcoming release

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Review by msJinLei — APPROVED** [review]

**Comment by msJinLei** [comment]
LGTM

File: src/Accounts/Accounts/ChangeLog.md [hunk 6]
```diff
@@ -19,6 +19,8 @@
 -->
 
 ## Upcoming Release
+* In `Connect-AzAccount` add warning message to deprecate username/password flow
+* In `Get-AzAccessToken` add Breaking Change notification to change AccessToken property from String to Secure String
```
Comment (isra-fel) [inline src/Accounts/Accounts/ChangeLog.md:23]:
```suggestion
* Added a warning message in `Connect-AzAccount` to discourage the use of the username/password (a.k.a ROPC) login flow.
* Preannounced a breaking change in `Get-AzAccessToken` to change `Token` property from `String` to `SecureString`.
``

**Review by isra-fel — CHANGES_REQUESTED** [review]

**Review by isra-fel — APPROVED** [review]

**Pull request merged**